### PR TITLE
Provide backwards compatible modelName implementation

### PR DESCRIPTION
**Pull request by mbutsko**
Replace deprecated and removed typeKey with modelName.

### CODE REVIEW

File: addon/mixins/attachable.js [hunk 1]
```diff
@@ -75,7 +75,7 @@ export default Ember.Mixin.create({
     }
   },
   _rootKey: function() {
-    return Ember.String.underscore(Ember.String.decamelize(this.constructor.typeKey));
+    return Ember.String.underscore(this._modelName);
```
Comment (esbanarango) [inline addon/mixins/attachable.js]: @mbutsko I think you forgot to add back the `Ember.String.decamelize` call here.

Comment (mbutsko) [inline addon/mixins/attachable.js]: In between the first and second request I noticed [in the release notes](http://emberjs.com/blog/2015/05/21/ember-data-1-0-beta-18-released.html) that:

> typeKey was previously always normalized to be a camelCased string.
> 
> In Ember Data 1.0.0-beta.18, this property is now called modelName. In addition, the 
> modelName is a dasherized string.

Which is why I removed the decamalize call.

Comment (esbanarango) [inline addon/mixins/attachable.js]: We should still keep it to support older versions.

File: addon/mixins/attachable.js [hunk 2]
```diff
@@ -16,7 +16,7 @@ export default Ember.Mixin.create({
     var adapter, attachmentKey, data, formData, promise, root, serializer, url,
     _this = this;
     adapter = this.store.adapterFor(this.constructor);
-    serializer = this.store.serializerFor(this.constructor.typeKey);
+    serializer = this.store.serializerFor(this._modelName);
```
Comment (esbanarango) [inline addon/mixins/attachable.js]: @mbutsko You're not calling the `_modelName` function anywhere. You're missing the `()` in front of the function name. Like this: `this._modelName()`. This is why all tests are failing.

**Pull request closed without merging**